library/core/src/intrinsics.rs

```diff
@@ -860,6 +860,9 @@ extern "rust-intrinsic" {
     /// cause [undefined behavior][ub] with this function. `transmute` should be
     /// the absolute last resort.
     ///
+    /// Transmuting pointers to integers in a `const` context is [undefined behavior][ub].
```

Comment:
my understanding is that transmuting pointers to integers is currently UB in *all* contexts, const-time or run-time.

Reply:
Well, nothing [in this list](https://doc.rust-lang.org/reference/behavior-considered-undefined.html) makes it UB, but it is also not explicitly allowed anywhere to my knowledge. It exists in the grey area of unspecified Rust. There are [reasons we might want to make it UB](https://github.com/rust-lang/unsafe-code-guidelines/issues/286), but I wouldn't know that there is T-lang consensus for that.

Reply:
Right, but isn't it the case that llvm already cannot perform the transmute properly and will already miscompile in some situations? Because if it's just plain broken we should just come clean about it, even if we do relax the rule again later once things are fixed.

Reply:
There are [theoretical miscompilations](https://github.com/rust-lang/unsafe-code-guidelines/issues/286#issuecomment-860189806), yes.